src/dotnet/commands/dotnet-build/IncrementalResult.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Tools.Build
+{
+    internal class IncrementalResult
+    {
+        public static IncrementalResult DoesNotNeedRebuild = new IncrementalResult(false, "", Enumerable.Empty<string>());
+
+        public bool NeedsRebuilding { get; }
+        public string Reason { get; }
+        public IEnumerable<string> Items { get; }
+
+        private IncrementalResult(bool needsRebuilding, string reason, IEnumerable<string> items)
+        {
+            NeedsRebuilding = needsRebuilding;
+            Reason = reason;
+            Items = items;
+        }
+
+        public IncrementalResult(string reason)
+            : this(true, reason, Enumerable.Empty<string>())
```

Comment:
(minor) This is totally a personal preference kind of thing, but just throwing it out there:

I like having the defaults of booleans to be 'false'.  It makes a very consistent experience for consumers of classes that take bools.  So usually I invert the name so the default value is false.  In this case "SkipRebuild".